rosbag2_transport/CMakeLists.txt

```diff
@@ -99,144 +100,153 @@ ament_export_libraries(${PROJECT_NAME})
-    test/rosbag2_transport/test_burst.cpp
-    INCLUDE_DIRS $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/rosbag2_transport>
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_record
-    test/rosbag2_transport/test_record.cpp
-    INCLUDE_DIRS $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/rosbag2_transport>
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common
-    ${SKIP_TEST})
-
-  rosbag2_transport_add_gmock(test_record_all
-    test/rosbag2_transport/test_record_all.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_record_all_ignore_leaf_topics
-    test/rosbag2_transport/test_record_all_ignore_leaf_topics.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_record_all_include_unpublished_topics
-    test/rosbag2_transport/test_record_all_include_unpublished_topics.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_record_all_no_discovery
-    test/rosbag2_transport/test_record_all_no_discovery.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_record_all_use_sim_time
-    test/rosbag2_transport/test_record_all_use_sim_time.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_keyboard_controls
-    test/rosbag2_transport/test_keyboard_controls.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_record_regex
-    test/rosbag2_transport/test_record_regex.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common
-    ${SKIP_TEST})
-
-  rosbag2_transport_add_gmock(test_record_services
-    test/rosbag2_transport/test_record_services.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_composable_recorder
-    test/rosbag2_transport/test_composable_recorder.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
-
-  rosbag2_transport_add_gmock(test_load_composable_components
-    test/rosbag2_transport/test_load_composable_components.cpp
-    LINK_LIBS rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common
-    ${rclcpp_components_TARGETS})
-
-  if(${rmw_implementation} MATCHES "rmw_cyclonedds(.*)")
-    ament_add_test_label(test_play_services__rmw_cyclonedds_cpp xfail)
-  endif()
+  ament_add_gmock_test(test_play
+    TEST_NAME test_play${target_suffix}
+    ENV ${rmw_implementation_env_var}
+    ${SKIP_TEST}
+  )
```

Comment:
@clalancette Can we somehow group invocation of the `ament_add_gmock_test(..) ` with the 
```
  ament_add_gmock_executable(test_play_loop test/rosbag2_transport/test_play_loop.cpp)
  target_link_libraries(test_play_loop rosbag2_transport ${test_msgs_TARGETS} rosbag2_test_common::rosbag2_test_common)
```
to land it in one place rather in two different locations in the CMake file?

Reply:
It's possible, but I think it is going to be tricky.  For instance, we could add each test to a list, and then have `create_tests_for_rmw_implementation` iterate over the list, registering each one.  But then we would also need a way to mark certain tests as `SKIP_TEST`, and it isn't clear to me how we would do that in a list.  Ideas?

Reply:
@clalancette I came across today the following approach
https://github.com/ros2/rclcpp/blob/2bb1dc2c716ae12a24145891724f8436aa4e52bd/rclcpp/test/rclcpp/CMakeLists.txt#L295-L315
```cpp
function(test_generic_pubsub_for_rmw_implementation)
  set(rmw_implementation_env_var RMW_IMPLEMENTATION=${rmw_implementation})
  ament_add_gmock(test_generic_pubsub${target_suffix} test_generic_pubsub.cpp
    ENV ${rmw_implementation_env_var}
  )
  if(TARGET test_generic_pubsub${target_suffix})
    target_link_libraries(test_generic_pubsub${target_suffix} ${PROJECT_NAME} rcl::rcl ${test_msgs_TARGETS})
  endif()
endfunction()
call_for_each_rmw_implementation(test_generic_pubsub_for_rmw_implementation)

function(test_qos_event_for_rmw_implementation)
  set(rmw_implementation_env_var RMW_IMPLEMENTATION=${rmw_implementation})
  ament_add_gmock(test_qos_event${target_suffix} test_qos_event.cpp
    ENV ${rmw_implementation_env_var}
  )
  if(TARGET test_qos_event${target_suffix})
    target_link_libraries(test_qos_event${target_suffix} ${PROJECT_NAME} mimick rcutils::rcutils rmw::rmw ${test_msgs_TARGETS})
  endif()
endfunction()
call_for_each_rmw_implementation(test_qos_event_for_rmw_implementation)
```
It doesn't form a list of tests for running with `call_for_each_rmw_implementation` but rather calls tests one by one.
But it least everything which belongs to the one test in the one place.

Reply:
All right, [1fce988](https://github.com/ros2/rosbag2/pull/1518/commits/1fce988cc6a4baf3bf78715a120e811929140377) does this.  It is a little verbose, but it gets the job done.  @MichaelOrlov please take another look when you get a chance.